lib/matplotlib/type1font.py

```diff
@@ -148,20 +148,20 @@ def _tokens(cls, text):
         """
         pos = 0
         while pos < len(text):
-            match = (cls._comment_re.match(text[pos:]) or
-                     cls._whitespace_re.match(text[pos:]))
+            match = (cls._comment_re.match(text, pos) or
+                     cls._whitespace_re.match(text, pos))
             if match:
                 yield (_TokenType.whitespace, match.group())
-                pos += match.end()
+                pos = match.end()
```

Comment:
We still have some `pos += ...` in some other branches. Are you sure that mixing absolute `pos = ...` and incremental `pos += ...` work together?

Reply:
If you're just reading one or two characters, then pos should just be incremented by 1 or 2.  On the other hand, when doing a regex (or substring) search, this PR makes it so that the regex machinery is passed the entire string (avoiding the substring construction), but additionally told to start searching from *pos*, so it will return a new *absolute* position.